File: microservice-transaction-sample/client/src/main/java/sample/client/command/GetOrdersCommand.java
package sample.client.command;

import io.grpc.ManagedChannel;
import java.util.concurrent.Callable;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;
import sample.rpc.GetOrdersRequest;
import sample.rpc.GetOrdersResponse;
import sample.rpc.OrderServiceGrpc;

@Command(name = "GetOrders", description = "Get order information by customer ID")
public class GetOrdersCommand implements Callable<Integer> {

  @Parameters(index = "0", paramLabel = "CUSTOMER_ID", description = "customer ID")
  private int customerId;

  @Override
  public Integer call() {
    ManagedChannel channel = Utils.getOrderServiceChannel();
    try {
      OrderServiceGrpc.OrderServiceBlockingStub stub = OrderServiceGrpc.newBlockingStub(channel);
      GetOrdersResponse response =
          stub.getOrders(GetOrdersRequest.newBuilder().setCustomerId(customerId).build());
      Utils.printJsonString(response);
      return 0;
    } catch (Exception e) {
      e.printStackTrace();
      return 1;
    } finally {
      Utils.shutdownChannel(channel);
    }
  }
}

File: dotnet/microservice-transactions-sample-with-shared-cluster-with-linq/Client/Commands/GetOrdersCommand.cs
using System.CommandLine;
using MicroserviceTransactionsSample.Client.Commands.Binders;
using MicroserviceTransactionsSample.Rpc;
using static MicroserviceTransactionsSample.Rpc.OrderService;

namespace MicroserviceTransactionsSample.Client.Commands;

public static class GetOrdersCommand
{
    private const string Name = "GetOrders";
    private const string Description = "Get orders information by customer ID";

    private const string ArgName = "customer_id";
    private const string ArgDescription = "customer ID";

    public static Command Create()
    {
        var customerIdArg = new Argument<int>(ArgName, ArgDescription);
        var getOrdersCommand = new Command(Name, Description)
        {
            customerIdArg
        };

        getOrdersCommand.SetHandler(getOrders,
                                    customerIdArg,
                                    new OrderServiceClientBinder());
        return getOrdersCommand;
    }

    private static async Task getOrders(int customerId, OrderServiceClient client)
    {
        var request = new GetOrdersRequest { CustomerId = customerId };
        var response = await client.GetOrdersAsync(request);

        Console.WriteLine(response);
    }
}
